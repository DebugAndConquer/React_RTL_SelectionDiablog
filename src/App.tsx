import './style/main.scss';
import SelectorWidget from './components/SelectorWidget';
import { ELEMENT_COUNT } from './components/helpers/variables';
import { generateSampleData } from './components/helpers/helpers';

const App = () => {
  return (
    <div className="main-container">
      <SelectorWidget options={generateSampleData(ELEMENT_COUNT)} />
    </div>
  );
}

export default App;
