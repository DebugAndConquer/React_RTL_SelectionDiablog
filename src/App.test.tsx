import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

test("Matches the snapshot", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
test("renders a component and verifies that main elements are displayed", () => {
  render(<App />);
  const headerElement = screen.getByText(/Element selection/i);
  const modalButtonElement = screen.getByText(/Change my selection/i);
  expect(headerElement).toBeInTheDocument();
  expect(modalButtonElement).toBeInTheDocument();
});
