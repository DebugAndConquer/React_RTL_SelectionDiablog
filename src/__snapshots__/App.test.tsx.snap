// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div>
  <div
    class="main-container"
  >
    <div
      class="sel-widget"
    >
      <h1>
        Element selection
      </h1>
      <p>
        Number of selected elements: 
        0
      </p>
      <div
        class="sel-widget__selected-opt-cont"
      >
        <p
          class="no-selected-options"
        >
          Selected elements will appear here
        </p>
      </div>
      <br />
      <button
        class="btn btn--green"
      >
        Change my selection
      </button>
    </div>
  </div>
</div>
`;
